syntaxes/vba.tmLanguage.json

```diff
@@ -27,63 +26,55 @@
 		"keywords": {
 			"patterns": [
 				{
-					"name": "keyword.$1.vba",
-					"match": "\\b(As|Attribute|Call|Const|Dim|Function|New|Nothing|Set|Sub)\\b"
+					"name": "keyword.conditional.vba",
+					"match": "(?i:\\b(Array|Else(If)?|For( Each)?|(I)?If|In|New|Then|To|Step|With)\\b)"
 				},
 				{
-					"name": "keyword.End.vba",
-					"match": "\\bEnd (Sub|Function)\\b"
+					"name": "keyword.conditional.end.vba",
+					"match": "(?i:\\bEnd (If|With)|Next\\b)"
 				},
 				{
-					"name": "keyword.control.$1.vba",
-					"match": "\\b(Each|For|In|Next|With)\\b"
+					"name": "keyword.control.vba",
+					"match": "(?i:\\b(As|And|By(Ref|Val)|Is|Like|Mod|Not|Optional|Or|Xor)\\b)"
 				},
 				{
-					"name": "keyword.control.End.vba",
-					"match": "\\bEnd (With)\\b"
+					"name": "keyword.$1.vba",
+					"match": "(?i:\\b(Attribute|End( (Function|Property|Sub))?|((Private|Public|Static) )?(Const|Function|Property|Sub))\\b)"
 				}
 			]
 		},
 		"numbers": {
 			"patterns": [
 				{
```

Comment:
I can understand you merged the number regexes to shorten them but it was a lot easier to understand their behaviour beforehand.
I'd prefer keeping them separated.